homeassistant/config.py

```diff
@@ -578,6 +578,44 @@ def find_annotation_rec(
     return find_annotation_rec(config, list(path), None)
 
 
+def stringify_invalid(ex: vol.Invalid) -> str:
+    """Stringify voluptuous.Invalid.
+
+    Based on voluptuous.error.Invalid.__str__, the main modification
+    is to format the path delimited by -> instead of @data[].
+    """
+    path = "->".join(str(m) for m in ex.path)
+    output = Exception.__str__(ex)
```

Comment:
```suggestion
    output = str(ex)
```

Reply:
That won't work because this function is meant as an alternative to the stringification done by `vol.Invalid.__str__`. I'll add a comment explaining this.

Reply:
We'll explain this further in a follow up.